### DISCUSSION TITLE
zip support

### DISCUSSION THREAD

**Discussion by axet** [audacious-media-player/audacious#563] [General]
Hello!

Can we have zip support for native playing from zip file without extracting?

Thanks!

**Comment by radioactiveman** [comment]
Zipping audio files makes no sense, you don't save disk space with it.
What's your use case and why do you not want to extract zip files?

**Reply by 4zbest** [reply]
Music from modarchive is typically distributed in zip format. It saves on average about 25% disk space. If audacious could handle zipped format it would save me hours of extracting files.

**Comment by axet** [comment]
Zipping audio files create proper collection from multiply files into one file. It add CRC checks to audio files. Instead of lot of directories you got just one file with your audiobook / album. Playing files from within zip add 0% CPU load, but improve collection management.

A lot of audio players support that feature: foobar, vlc

**Comment by birdie-github** [comment]
Playing compressed files without decompressing them first is possible but seeking, AFAIK, not so much.

And without seeking this whole feature request becomes quite moot.

**Reply by axet** [reply]
Zip support seeking if opened as file.

**Reply by birdie-github** [reply]
This statement makes no sense. You don't understand compression unfortunately.

**Reply by axet** [reply]
Don't be silly.

Zip allows to open arbitrary entry regardless of order, if zip file opened as file and not as pipe (stream). Piped zip, does not allow open arbitrary entries.

Working with zip stream (archived data) can be organized as 1) caching by caching media (mp3) headers of current stream or simply having big memory cache of 1-2 MB or 2) simply reopening stream as new and skipping to required offset when seek backward.

**Reply by birdie-github** [reply]
Patches are welcome.

**Reply by birdie-github** [reply]
Good luck opening 75GB zipped files that way (seen them in the wild). Got enough RAM? or SWAP space? Prepared to wait an hour to have this file "cached" whatever that means?

**Comment by axet** [comment]
Seesm like this feature already been partially implemented in archive_reader.cc

**Reply by EarBiteR** [reply]
On linux I use ziprofs.py to mount a folder of zips to let me access everything in them transparently without having to unzip.  You could do the same for your album zips. -> https://github.com/openscopeproject/ZipROFS

**Reply by axet** [reply]
Nautilus Archive Mounter does the job just fine. No need to install anything else. I know. But prefer native support.

Right now I'm using my python script, which allow to play zip content and even calculate total time of zip albums on my HDD. I guess CPP app can do the same even better.

**Reply by EarBiteR** [reply]
The beauty of ziprofs is that I dont have to mount any zips. I mount the folder of zips and access all contents navigating through them at will from any APP!  :-)  I thought with Nautilus as well as Nemo ( I run mint/cinnamon)  you still have to 'mount' the archive as a disk to access it.  I guess if you have it all in one zip that works well for you.  I have thousands though..
